src/hooks/useDeck.js

```diff
@@ -18,35 +24,96 @@ function useDeck(initialState, numSlides, looping, animationsWhenGoingBack) {
           }
           return { ...state };
         }
         return action.type === 'NEXT_SLIDE_IMMEDIATE'
-          ? { currentSlide: state.currentSlide + 1, immediate: true }
-          : { currentSlide: state.currentSlide + 1 };
+          ? {
+              currentSlide: state.currentSlide + 1,
+              immediate: true,
+              currentSlideElement: 0,
+              immediateElement: false,
+              reverseDirection: false
+            }
+          : {
+              currentSlide: state.currentSlide + 1,
+              currentSlideElement: 0,
+              immediate: false,
+              immediateElement: false,
+              reverseDirection: false
+            };
       case 'PREV_SLIDE':
-        // If current slide is inital slide then if looping go
+        // If current slide is initial slide then if looping go
         // to last slide else stop
         if (state.currentSlide === initialState.currentSlide) {
           if (looping) {
             return {
               currentSlide: numSlides - 1,
-              immediate: animationsWhenGoingBack ? true : false
+              currentSlideElement: Math.max(slideElementMap[numSlides - 1], 0),
+              immediate: !!animationsWhenGoingBack,
+              reverseDirection: true
             };
           }
           return { ...state };
         }
         return {
           currentSlide: state.currentSlide - 1,
-          immediate: animationsWhenGoingBack ? true : false
+          currentSlideElement: Math.max(
+            slideElementMap[state.currentSlide - 1],
+            0
+          ),
+          immediate: !!animationsWhenGoingBack,
+          immediateElement: true,
+          reverseDirection: true
+        };
+      case 'NEXT_SLIDE_ELEMENT': {
+        return {
+          ...state,
+          currentSlideElement: state.currentSlideElement + 1,
+          immediateElement: false,
+          reverseDirection: false
+        };
+      }
+      case 'NEXT_SLIDE_ELEMENT_IMMEDIATE': {
+        return {
+          ...state,
+          currentSlideElement: state.currentSlideElement + 1,
+          immediateElement: true,
+          reverseDirection: false
+        };
+      }
+      case 'PREV_SLIDE_ELEMENT': {
+        return {
+          ...state,
+          currentSlideElement: Math.max(state.currentSlideElement - 1, 0),
+          immediateElement: false,
+          reverseDirection: true
+        };
+      }
+      case 'PREV_SLIDE_ELEMENT_IMMEDIATE': {
+        return {
+          ...state,
+          currentSlideElement: Math.max(state.currentSlideElement - 1, 0),
+          immediateElement: true,
+          reverseDirection: true
         };
+      }
       default:
         return { ...state };
     }
   }
-  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const [state, dispatch] = React.useReducer((state, action) => {
+    const _reducer = reducer(state, action);
+    // console.log(_reducer);
```

Comment:
This is logger middleware for the time being and is useful for development and debugging. It will be removed come release time.